api/client/proto/authservice.proto

```diff
@@ -780,12 +786,48 @@ message DeleteMFADeviceRequestInit {
 // DeleteMFADeviceResponseAck is a confirmation of successful device deletion.
 message DeleteMFADeviceResponseAck {}
 
-// GetMFADeviceRequest is a request for MFA devices for the calling user.
-message GetMFADevicesRequest {}
+// GetMFADevicesRequest is a request for MFA devices for the calling user.
+message GetMFADevicesRequest {
+    // Request defines how the user will be verified before receiving list of devices.
+    oneof Request {
+        // TokenID verifies existence of token and retrieves devices for user defined in token.
+        string TokenID = 1;
```

Comment:
Why do we need the token here? Didn't this already work without a token?

Reply:
> Didn't this already work without a token?

Yes that was before we needed user to view their devices without being logged in. I did originally create a seperate endpoint called `GetMFADevicesWithToken` but b/c we wanted to reduce mfa api surface i combined it.

Reply:
> It escaped me that they may list devices with a token without being logged in. Is this true for all token uses? Let's get this clearly documented in all places where it is applicable.

Most of user tokens like reset, invite, and recovery are for users who are not logged in. The only exception is the privilege token (which isn't introduced until later PR)

> Also, why do we need the AuthenticatedUser field? Don't we know from the context if they are authenticated?

Revised, i saw we can get the username from context like this: https://github.com/gravitational/teleport/blob/master/lib/auth/permissions.go#L717 (without setting a default name if username is empty)